app.rb

```diff
@@ -38,18 +46,23 @@ def auth_server_uri
       session[:agency] = agency
       erb :"agency/#{agency}/index", layout: false, locals: { logout_msg: logout_msg }
     else
-      ial = params[:ial] || '1'
-      aal = params[:aal] || '1'
+      ial = sanitize(params[:ial]) || '1'
+      aal = sanitize(params[:aal]) || '1'
```

Comment:
Would it be better to validate that `params[:ial/:aal]` is in a list of acceptable values instead of sanitizing?
Could be re-used in the view as well:

https://github.com/18F/identity-saml-sinatra/blob/master/views/index.erb#L96-L117

Reply:
My thinking was that we kind of have all the acceptable values scattered all over in this app, didn't think it was worth the effort to unify them all. How strongly do you feel?

Reply:
Not terribly strongly 🙂

Reply:
I'd tend to agree with @mitchellhenke , if we can know them to be of a fixed set of allowable values, it feels much safer a guarantee than to try to sanitize an arbitrary user input. To your point about scattered acceptable values, could we at least have a predefined set here, even if it's also duplicated elsewhere in the app? Or is it that we don't know what all values might exist?

Reply:
Gave that a shot in 6269c3b946b63ea46f0ba22dd46b05865f1cf72d

Reply:
Looks good to me!